SwissTransferFeatures/SettingsView/EditSettings/NotificationsSettings.swift

```diff
@@ -0,0 +1,155 @@
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakDI
+import STCore
+import STResources
+import SwiftUI
+import SwissTransferCore
+import SwissTransferCoreUI
+
+enum NotificationsSettingsModel: Hashable, CaseIterable {
+    case allNotifications
+    case newTransfers
+    case downloadsInProgress
+    case finishedTransfers
+    case downloadTransfers
+    case failedTransfers
+    case expiredTransfers
+
+    var localized: String {
+        switch self {
+        case .allNotifications:
+            STResourcesStrings.Localizable.settingsAllNotifications
+        case .newTransfers:
+            STResourcesStrings.Localizable.settingsNotificationsTransferInProgress
+        case .downloadsInProgress:
+            STResourcesStrings.Localizable.settingsNotificationsDownloadInProgress
+        case .finishedTransfers:
+            STResourcesStrings.Localizable.settingsNotificationsTransferFinished
+        case .downloadTransfers:
+            STResourcesStrings.Localizable.settingsNotificationsDownloadTransfer
+        case .failedTransfers:
+            STResourcesStrings.Localizable.settingsNotificationsTransferNotDownloaded
+        case .expiredTransfers:
+            STResourcesStrings.Localizable.settingsNotificationsLinkExpired
+        }
+    }
+}
+
+public enum DefaultPreferences {
+    public static let notificationsNewTransfers = true
+    public static let notificationsDownloadInProgress = true
+    public static let notificationsFinishedTransfers = true
+    public static let notificationsDownloadTransfers = true
+    public static let notificationsFailedTransfers = true
+    public static let notificationsExpiredTransfers = true
+}
+
+struct NotificationsSettings: View {
+    @AppStorage(UserDefaults.shared.key(.notificationsNewTransfers))
+    private var newTransfers = DefaultPreferences.notificationsNewTransfers
+
+    @AppStorage(UserDefaults.shared.key(.notificationsDownloadInProgress))
+    private var downloadInProgress = DefaultPreferences.notificationsDownloadInProgress
+
+    @AppStorage(UserDefaults.shared.key(.notificationsFinishedTransfers))
+    private var finishedTransfers = DefaultPreferences.notificationsFinishedTransfers
+
+    @AppStorage(UserDefaults.shared.key(.notificationsDownloadTransfers))
+    private var downloadTransfers = DefaultPreferences.notificationsDownloadTransfers
+
+    @AppStorage(UserDefaults.shared.key(.notificationsFailedTransfers))
+    private var failedTransfers = DefaultPreferences.notificationsFailedTransfers
+
+    @AppStorage(UserDefaults.shared.key(.notificationsExpiredTransfers))
+    private var expiredTransfers = DefaultPreferences.notificationsExpiredTransfers
+
+    @State private var allNotificationsEnabled = true
+
+    init() {
+        allNotificationsEnabled = computeAllNotificationsEnabled()
+    }
+
+    var body: some View {
```

Comment:
issue: Disabling "all notifications" doesn't disable the other notifications

Reply:
Fixed this morning. Pushing soon.

Reply:
Are we sure this is the behaviour we want ? Mail does not work like this, as we checked together. (I know Mail is a bit different also but arguably the same.)

Reply:
Great question! This is a tricky part of the settings 😅
We will discuss it tomorrow!

Reply:
Implemented the desired behaviour. I managed with only one extra bool.